### PR TITLE
TEST: Add an algorithm ID tester for libcrypto vs provider

### PR DESCRIPTION
**Pull request by levitte**
Providers produce algorithm IDs of their own, and we need to compare
them against the same thing produced by libcrypto's ASN.1 code and
with legacy keys.

This tester can compare algorithm IDs for signatures and for keys,
given certificates that hold such data.

To verify key algorithm IDs, only one certificate is necessary, and
its public key is used.

To verify certificate algorithm IDs, we need to launch the signature
operation that would verify a certificate against the public key of
its signing CA, so that test needs two files.

### CODE REVIEW

File: test/recipes/06-test_algorithmid.t [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     'ee-cert2' => 'ca-cert2',
+     'ee-cert-ec-explicit' => 'ca-cert-ec-named',
+     'ee-cert-ec-named-explicit' => 'ca-cert-ec-explicit',
+     'ee-cert-ec-named-named' => 'ca-cert-ec-named',
+     # 'ee-pss-sha1-cert' => 'ca-cert',
+     # 'ee-pss-sha256-cert' => 'ca-cert',
+     # 'ee-pss-cert' => 'ca-pss-cert',
+     'server-ecdsa-brainpoolP256r1-cert' => 'rootcert',
+     'server-pss-restrict-cert' => 'rootcert',
+     # 'server-ed448-cert' => 'root-ed448-cert'
```
Comment (levitte) [inline test/recipes/06-test_algorithmid.t:30]: The reason some of these are currently commented away is that `EVP_DigestVerifyInit()` fails for ED448, and that there's a mismatch between the RSASSA-PSS signature in ee-pss certs and the acceptable signature algorithm given from the EVP_KEYMGMT...  That needs deeper coding to resolve, and wll *not* be part of this PR.

**Comment by levitte** [comment]
The CI failures here should be fixed as soon as #14030 is merged.  As a matter of fact, I'll merge this into #14030 to prove that point.

**Review by t8m — APPROVED** [review]
Approved as the failures will be fixed when this is rebased on top of #14030

**Comment by levitte** [comment]
Merged

388eb0d9709b4edf0fe4edf207b23d924fde2649 TEST: Add an algorithm ID tester for libcrypto vs provider

**Pull request closed without merging**